src/internal/connector/onedrive/drive.go

```diff
@@ -236,10 +236,20 @@ func getFolder(
 	rawURL := fmt.Sprintf(itemByPathRawURLFmt, driveID, parentFolderID, folderName)
 	builder := msdrive.NewItemsDriveItemItemRequestBuilder(rawURL, service.Adapter())
 
-	foundItem, err := builder.Get(ctx, nil)
+	var (
+		foundItem   models.DriveItemable
+		err         error
+		internalErr error
+	)
+
+	err = graph.RunWithRetry(func() error {
+		foundItem, internalErr = builder.Get(ctx, nil)
+		return internalErr
+	})
```

Comment:
Trying to juggle internalErr and err is an anti-pattern which will cause unexpected issues.  This is why `RunWithRetry` returns the final outcome error (ie, the last occurring error), so that the caller doesn't need to handle this behavior.
```suggestion
	var (
		foundItem   models.DriveItemable
		err         error
	)

	err = graph.RunWithRetry(func() error {
		foundItem, err = builder.Get(ctx, nil)
		return err
	})
```